### ISSUE TITLE
Qt Creator crashes when using the plugin in 8.0.1

### ISSUE THREAD

**Issue by docquantum** [CJCombrink/SpellChecker-Plugin#140]
When I try to use the latest release for 8.0.0 on 8.0.1, Qt Creator crashes when trying to access the settings of the spell checker or load up a project.

Qt Creator seems to load the plugin fine and doesn't show any errors.

**Comment by CJCombrink** [comment]
@docquantum Thanks for reporting

I can reproduce on Windows. 

Perhaps the following might also be able to assist:
- What OS are you using?

On my side it seems like Spell Checking in general is still working when I edit files. It just crashes when I open the settings. Is that also what you see?

**Comment by docquantum** [comment]
I am on Ubuntu Linux 22.04. Qt Creator was installed through the Qt Installer.

To be more specific, with the plugin loaded, if I tried to access the settings or open a project, Qt Creator would crash.

Thanks for the help.

**Comment by CJCombrink** [comment]
Since it happens on Windows as well it is somewhere between the plugin and Qt Creator, I doubt that it is a configuration issue.

I have logged a ticket on the Qt Creator bug tracker under [QTCREATORBUG-28075](https://bugreports.qt.io/browse/QTCREATORBUG-28075) for them to look into this.
I have also started a build using the latest version of Qt to build the plugin.

For now I am unable to do more until tonight after work (in about 12/13 hours)

**Comment by CJCombrink** [comment]
@docquantum 

I have updated my build action to use the same version of Qt that the installer installed Qt Creator uses.
On my personal Windows machine it seemed to resolve the issue.

Can you please test and let me know.
The downloads can be found [here](https://github.com/CJCombrink/SpellChecker-Plugin/actions/runs/2877424039), at the bottom of the page.
Just take note, for some reason the archives from the actions might need to be extracted twice (Archive in an archive)

**Comment by docquantum** [comment]
@CJCombrink 

Thanks for working on it.

Installing the new plugin did not seem to fix the issue on my end.

The same behavior is observer. If I run QtCreator through a terminal, the log shows it segfaults when accessing the settings:
```
~ % ./Qt/Tools/QtCreator/bin/qtcreator.sh
error: : cannot open
error: : cannot open
error: : cannot open
loadUserAddedWords: User dictionary name empty
[1]    271465 segmentation fault (core dumped)  ./Qt/Tools/QtCreator/bin/qtcreator.sh
```

The version is of the plugin is 3.1.0 according to QtCreator.

**Comment by CJCombrink** [comment]
Perhaps it can be due to the configuration issue since it is first install.

With Qt Creator closed: 
1. Browse to the location/folder of the Qt Creator settings file (QtCreator.ini)
    - See the [Location of Settings Files](https://doc.qt.io/qtcreator/creator-quick-tour.html#location-of-settings-files)
1. Make a backup or copy of the file
1. Open the file in a text editor
2. Delete the `[SpellCheckerPlugin]` section and everything underneath it.
    - On my Windows system it is about 24 lines but can be more or less. 
4. Save the file and close
5. Run Qt Creator and see if you can access the settings page.

If the above does not improve the situation I unfortunately will have to get a Linux install going to debug further.

**Comment by docquantum** [comment]
Unfortunately, even after modifying the config file and removing all instances of `SpellChecker`, then restarting QtCreator, it would still crash.

**Comment by CJCombrink** [comment]
Sorry for the delay in getting this fixed.
I am actively working on this when I have time. Between family responsibilities and a broken keyboard I am making progress. I have a Linux environment that I could reproduce, now I just need to find the cause and fix.

**Comment by CJCombrink** [comment]
If you have a moment please the attached. I have some confidence that it should work
[SpellChecker-Linux.zip](https://github.com/CJCombrink/SpellChecker-Plugin/files/9386622/SpellChecker-Linux.zip)

I am busy building similar in the actions. If they work I will create a new release

**Comment by zzhiyi** [comment]
I can confirm it's fixed now. Thank you for your work!

**Comment by docquantum** [comment]
I too can confirm that it's fixed. I no longer am crashing and am able to access the settings menu.

Thank you for your work!

I'll leave it up to you to close this issue when you feel everything has been resolved.

**Comment by CJCombrink** [comment]
@docquantum Thank you for the feedback.
The Qt Creator bug I logged also confirmed that they made a binary compatibility breaking change which caused the issue.
Just glad it is resolved now.
